src/server/debug/server/server.go

```diff
@@ -283,13 +286,13 @@ func (s *debugServer) Profile(request *debug.ProfileRequest, server debug.Debug_
 		collectProfileFunc(request.Profile),
 		nil,
 		nil,
-		redirectProfileFunc(pachClient.Ctx(), request.Profile),
+		redirectProfileFunc(request.Profile),
 		collectProfileFunc(request.Profile),
 	)
 }
 
 func collectProfileFunc(profile *debug.Profile) collectFunc {
-	return func(tw *tar.Writer, prefix ...string) error {
+	return func(_ context.Context, tw *tar.Writer, _ *client.APIClient, prefix ...string) error {
 		return collectProfile(tw, profile, prefix...)
```

Comment:
`collectProfile` calls `debugProfile` with a callback which calls `writeProfile`, which calls `time.Sleep`; it makes sense to propagate the context down the chain so that `writeProfile` can terminate the sleep early if the context is cancelled.

Reply:
Done.